nncf/quantization/passes.py

```diff
@@ -10,60 +10,55 @@
 # limitations under the License.
 
 import collections
-from typing import List, Optional, TypeVar
+from typing import List, TypeVar
 
 from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.graph import NNCFNode
 from nncf.common.graph.operator_metatypes import OperatorMetatype
 
 TModel = TypeVar("TModel")
 
 
 def transform_to_inference_graph(
     nncf_graph: NNCFGraph,
+    input_nodes: List[NNCFNode],
     shapeof_metatypes: List[OperatorMetatype],
     dropout_metatypes: List[OperatorMetatype],
```

Comment:
We've added `input_nodes` as input. Why not add `ShapeOf`, `Dropout` etc. nodes in the same way?
Also, the change looks not necessary cause not improving any aspect of the method. Or is it improved?

Reply:
Only improvement is that `input_nodes` are chosen inside the backend-specific code. This allows to use common `remove_shapeof_subgraphs` and  `filter_constant_nodes` for graphs with several connected components.